### ISSUE TITLE
(Feature Request) Create Flask App

### ISSUE THREAD

**Issue by marshalmiller** [rottingresearch/linkrot#3]
https://realpython.com/python-web-applications/

**Comment by aditirao7** [comment]
Could I work on this? I am familiar with deploying flask apps on heroku.

**Comment by marshalmiller** [comment]
@aditirao7 amazing! Yes. Please. This is something I have been putting off for a while so if you could be of assistance, I'd really appreciate it.

**Comment by aditirao7** [comment]
Alright, I'll get started on it right away 😄. Is there a specific UI design you have in mind or should I keep it simple for now?

**Comment by marshalmiller** [comment]
Just start simple but go ahead and put any spin you want on it. I have nothing in mind.

**Comment by aditirao7** [comment]
@marshalmiller I am having some trouble with the linkrot package, I'll attach a screenshot. It installed successfully but none of the commands work.
![Screenshot from 2021-10-03 14-10-10](https://user-images.githubusercontent.com/57457307/135746270-5f4f10c9-19e4-4b79-b1de-33527634e99f.png)

**Comment by marshalmiller** [comment]
Oh. I apologize. The version on the pip repository is broken. I will fix it ASAP. I will reach out again once it is fixed. I'm sorry about that. Thank you for bringing it to my attention.

**Comment by aditirao7** [comment]
@marshalmiller No problem, do let me know when it's fixed. I deployed the app temporarily [here](https://linkrot-temp.herokuapp.com/) so you can monitor the progress.

**Comment by marshalmiller** [comment]
@aditirao7 All good now. Just make sure you install as sudo. You'll get a PATH error if not. Please let me know if you have any more issues or questions.

**Comment by aditirao7** [comment]
@marshalmiller Thanks that worked, the exact command I had to use was: ```sudo -H pip3 install linkrot```
Not using the H flag and using pip instead of pip3 gave errors. I could update the ReadME if you want?

**Comment by marshalmiller** [comment]
Thank you. I appreciate that. Let me work on the installation method. I know there are a few bugs swimming around there anyway. You're already doing so much too. I cannot thank you enough. That draft you shared looks awesome.

**Comment by marshalmiller** [comment]
@aditirao7 If you find anything else that isn't streamlined or accurate to documentation like that, please feel free to open an issue too. You've been so much help.

**Comment by aditirao7** [comment]
@marshalmiller Will do :smile:! Can linkrot be used as a python library or is it only terminal commands as of now? Maybe that functionality can be added?

**Comment by marshalmiller** [comment]
@aditirao7 I certainly don't have anything against using it that way. There's a chance it works right now. I've never fully vetted it because I was putting off the Flask App, but given how it's published, it might actually work as-is. Please let me know if there is an error report or if you run into problems. I might have time later to check myself.

**Comment by aditirao7** [comment]
@marshalmiller Oh! Okay, I'll look into the code. I was trying to use it this way:

![Screenshot from 2021-10-03 23-05-23](https://user-images.githubusercontent.com/57457307/135765128-3b690dfb-893c-4411-aa2b-93b17cdd9aca.png)

But formatting the report after sending it to the frontend is a bit tedious, hence why I asked about the library :sweat_smile:. I can either try to import the library or do some text cleaning in python before sending it to HTML. Will keep you posted.

**Comment by aditirao7** [comment]
@marshalmiller 
I found the documentation for the library part in the comments of [__init__.py](https://github.com/marshalmiller/linkrot/blob/main/linkrot/__init__.py). I opened an issue for adding that to the main documentation. (#27)
The [web app](https://linkrot-temp.herokuapp.com/) is done, for the most part. 

What's left:
1. The report for the broken and active links takes quite a long time to generate through the terminal command. I'll look through more of the code tomorrow to check if the library has a function for it (might be faster). I'll add the downloading option once this is sorted.
2. The design of the analysis page could be better. 
3. The about page needs content.
4. Deploying the web app at a permanent URL.

**Comment by marshalmiller** [comment]
@aditirao7 Awesome. I'm glad you found it. I will make sure to address #27. Obviously Tasks 3 & 4 I can take care of. I have a space to host this. Do you think you could share the app in the repo? Maybe a Flask branch?

**Comment by aditirao7** [comment]
@marshalmiller Yep will open a PR, was waiting for you to take a look. Should I make a separate Flask branch because there's already one called Flask-App?

**Comment by marshalmiller** [comment]
@aditirao7 Oh. No. Please use the one that is already created. Thank you so much. It's looking great.

**Comment by aditirao7** [comment]
@marshalmiller Sorry the code is a bit messy right now but I'll add comments once everything is done.

**Comment by marshalmiller** [comment]
@aditirao7 Since the code is so different from the initial project and we have the project working as a library, I have created a separate project just for the Flask App. If you don't mind, can we move our work over to that repo?
[https://github.com/marshalmiller/rottingresearch](https://github.com/marshalmiller/rottingresearch)

**Comment by aditirao7** [comment]
Alright cool, I'll make all new PR's to that repo.

**Comment by marshalmiller** [comment]
Perfect! Thank you. You're so much help. Thank you for being so helpful and easy to work with. I'm going to close this issue since we are moving over to the new project.
